File: p7.py
import random
from matplotlib import pyplot as plt
observed =[1.88,0.54,1.9,0.15,0.02,2.81,1.50,1.53,2.62,2.67, 3.53,0.53,1.80,1.80,0.79,0.21,0.8,0.26,0.63,0.36,2.03,1.42,1.28,0.82,2.16,0.05,0.04,1.49,0.66,0.2,0.3,1,0.34,0.34,0.01,0.10,1.10,0.24,0.26,0.45,0.17,4.29,0.80,5.50,4.91,0.35,0.36,0.90,1.03,1.73,0.38,0.48]
expected = []
for i in range(len(observed)):
	observed[i] = int(observed[i]/10)
alpha = 0.05
df = 9
#print("How many random numbers you want to generate?")
num = len(expected) #int(input())
#print("How many intervals?")
interval = 10#int(input())
for i in range(0,interval):
	#observed.append(0)
	expected.append(int(sum(observed)/(interval+1)))
total = 0
for i in range(0,10):
	diff = observed[i] - expected[i]
	diff = diff * diff
	total = total + (diff/(expected[i]+1))
print("chi square sum: ",total)
print("alpha: ",alpha)
print("Degree of freedom: ",df)
print("Chi square at (",alpha,",",df,"): 16.9")
if total < 16.9:
	print("Accepted")
else:
	print("Rejected")
x = []
for i in range(10):
	x.append(0)
for i in observed:
	if i <0.1:
		x[0]+=1
	elif i<0.2:
		x[1]+=1
	elif i<0.3:
		x[2]+=1
	elif i<0.4:
		x[3]+=1
	elif i<0.5:
		x[4]+=1
	elif i<0.6:
		x[5]+=1
	elif i<0.7:
		x[6]+=1
	elif i<0.8:
		x[7]+=1
	elif i<0.9:
		x[8]+=1
	else:
		x[9]+=1
plt.plot(x,[i for i in range(10)])
plt.show()

File: p7.java
import java.util.Scanner;
public class p7{
	public static void main(String args[])
	{
		Scanner in=new Scanner(System.in);
		System.out.println("Enter number of random numbers to be generated");
		int n=in.nextInt();
		int arr[] = new int[10];
		for(int i=0;i<arr.length;i++)
    			arr[i] = 0;
		for(int i=0;i<n;i++)
		{
			double val=(Math.random()*10);
			if(val<1)
				arr[0]++;
			else if(val>=1 && val<2)
				arr[1]++;
			else if(val>2 && val<3)
				arr[2]++;
			else if(val>=3 && val<4)
				arr[3]++;
			else if(val>=4 && val<5)
				arr[4]++;
			else if(val>=5 && val<6)
				arr[5]++;
			else if(val>=6 && val<7)
				arr[6]++;
			else if(val>=7 && val<8)
				arr[7]++;
			else if(val>=8 && val<9)
				arr[8]++;
			else if(val>=9 && val<=10)
				arr[9]++;		
		}
		Draw(arr);
	       int E=(int)(n/9);
		double sum=0;
		for(int i=0;i<10;i++)
		{
			System.out.println("Observed : "+arr[i]+" Expected value: "+E);
			int r=(int)Math.pow((arr[i]-E),2);
			double o = (double)r/E;			
			sum=sum+o;
			System.out.println("Chi square for "+ (i+1) +" is "+o);
			System.out.println("------------------------------");		
		}
		System.out.println("Final chi square sum "+sum);
		double alpha=0.05;
		double H0=16.910;
		if(sum>H0)
			System.out.println("Rejected as sum "+sum+" is greater than H0 "+H0+" ");
		else
			System.out.println("\nAccepted.Sum "+sum+" is less than H0 "+H0+" ");
	}
		public static void Draw(int arr[])
		{
			int max=0;
			int arr1[]=new int[arr.length];
			for(int i=0;i<arr.length;i++)
			{
				arr1[i]=arr[i];
				if(arr[i]>max)
					{max=arr[i];}
			}
			System.out.println("Graph with frequency on y axis from 0 to "+max+" and classes on x axis");
			System.out.println("---------------------------------------------------------------------");
			for(int i=max;i>0;i--)
			{
				//System.out.print(max);
				for(int j=0;j<arr1.length;j++)
				{	
					if(arr1[j]==max)
					 {
						arr1[j]--;
						System.out.print(" I  ");
					}
					else{
						System.out.print("    ");
					}
				}
				System.out.println("");
				max--;
			}
			System.out.println("0-1 1-2 2-3 3-4 4-5 5-6 6-7 7-8 8-9 9-10");
		}
		
}

